tests/Support/CacheStoreTest.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Spatie\Ray\Tests\Support;
+
+use PHPUnit\Framework\TestCase;
+use Spatie\Ray\Support\CacheStore;
+use Spatie\Ray\Tests\TestClasses\ClockMock;
+
+class CacheStoreTest extends TestCase
+{
+    /** @var ClockMock */
+    private $clock;
+
+    /** @var CacheStore */
+    private $store;
+
+    public function setUp(): void
+    {
+        $this->clock = new ClockMock();
+        $this->store = new CacheStore($this->clock);
+    }
+
+    /** @test */
+    public function it_can_count_per_seconds(): void
```

Comment:
Here you're testing the implementation of clock, stores, ...

I think it might be better to test the entire behaviour.

So set a rate limit of 3, try to do 4 ray calls, and assert that only three were sent. This way if we change something about the way we implement the rate limit, the tests will still pass.

Reply:
this is covered in RayTest